File: assignment-3-crud-practice-Jwizzed-main/PeopleRecordManagement.py
from StudentXMLRecorder import StudentXMLRecorder
from Person import Person
from Student import Student


class PeopleRecordManagement:
    def __init__(self):
        self.peopleXMLRecorder = StudentXMLRecorder()
        self.peopleList = self.peopleXMLRecorder.loadStudentData()

    def add(self, record):
        if (not self.find(record.idNumber)) and (record.idNumber != 0):
            self.peopleList.append(record)
            self.peopleXMLRecorder.saveStudentData(self.peopleList)
        else:
            raise ValueError("Record already exists in the Record list or Invalid ID number")


    def find(self, idNumber: int):
        for people in self.peopleList:
            if people.idNumber == idNumber:
                return True
        return False

    def delete(self, recldNumber: int):
        for people in self.peopleList:
            if people.idNumber == recldNumber:
                self.peopleList.remove(people)
                print("Record Deleted Successfully")
                try:
                    self.peopleXMLRecorder.saveStudentData(self.peopleList)
                except Exception:
                    continue

    def findRecord(self, name:str, idNumber:int, contactNumber:str):
        choose = None
        for people in self.peopleList:
            if people.idNumber == idNumber and people.name == name and people.contactNumber == contactNumber:
                choose = people
        if choose is None:
            raise ValueError("Record not found")
        return choose

    def update(self, name:str, idNumber: int, contactNumber: str):
        if self.find(idNumber):
            rec = self.findRecord(name, idNumber, contactNumber)
            name = input("What is the new Student Name? ")
            idNumber = int(input("What is the new Student id Number? "))
            contactNumber = input("What is the new Student contact Number? ")


            try:
                test_rec = Person(name, idNumber, contactNumber)
                self.peopleList.remove(rec)
                rec = Student(name, idNumber, contactNumber)
                self.peopleList.append(rec)
                self.peopleXMLRecorder.saveStudentData(self.peopleList)
                print("Record Updated Successfully")

            except ValueError as e:
                print(e)

        else:
            print("Record Not Found in the Student list")

    def display(self, sort_by: int):
        display_list = self.peopleList
        if not self.peopleList:
            return ["No records found"]
        if sort_by == 1:
            display_list.sort(key=lambda x: x.name)
        elif sort_by == 2:
            display_list.sort(key=lambda x: x.idNumber)
        elif sort_by == 3:
            display_list.sort(key=lambda x: x.contactNumber)
        return display_list

File: assignment-5-Jwizzed-main/PeopleRecordManagement.cs
namespace ConsoleApplication1;

public class PeopleRecordManagement
{
    private LinkedList<Person> peopleList;

    public LinkedList<Person> getPeopleList()
    {
        return peopleList;
    }

    public PeopleRecordManagement()
    {
        peopleList = new LinkedList<Person>();
    }

    public void add(Person record)
    {
        if (!find(record.getIdNumber()) && (record.getIdNumber() != 0))
        {
            peopleList.AddLast(record);
        }
        else
        { 
            throw new Exception("Record already exists in the Record list or Invalid ID number");
        }
    }
    
    public bool find(int idNumber){
        foreach (Person people in peopleList){ //for loop to find the match number
            if (people.getIdNumber() == (idNumber)){
                return true;
            }
        }return false;
    }
    
    public void delete(int recldNumber){
        foreach (Person people in peopleList){//for loop to find the recldNumber
            if (people.getIdNumber() == (recldNumber)){
                peopleList.Remove(people);
                break;
            }
        } 
    }
    public Person findRecord(string name, int idNumber, string contactNumber){
        Person choose = new Person();
        foreach (Person people in peopleList){//for loop to find the idNumber
            if (people.getIdNumber() == (idNumber)){
                choose = people;
            }
        }return choose;
    }
    
    public void update(string name, int idNumber, string contactNumber) {
        if (find(idNumber)) {
            Person rec = findRecord(name, idNumber, contactNumber);
            
            Console.WriteLine("What is the new Student Name ? ");
            name = Console.ReadLine();
            
            Console.WriteLine("What is the new Student id Number ? ");
            idNumber = Convert.ToInt32(Console.ReadLine());

            // Display message only
            Console.WriteLine("What is the new Student contact Number ");
            contactNumber = Console.ReadLine();

            try{
                Person test_rec = new Person();

                test_rec.setIdNumber(idNumber);
                test_rec.setName(name);
                test_rec.setContactNumber(contactNumber);

                peopleList.Remove(rec);
                rec.setIdNumber(idNumber);
                rec.setName(name);
                rec.setContactNumber(contactNumber);
                peopleList.AddLast(rec);

                Console.WriteLine("Record Updated Successfully");

            }catch (Exception e) {
                Console.WriteLine(e.Message);
            }
        }
        else {
            // Print statement
            Console.WriteLine("Record Not Found in the Student list");
        }
    }

    public LinkedList<Person> display(int sortBy)
    {
        LinkedList<Person> display_list = peopleList;
        if (!display_list.Any())
        {
            return new LinkedList<Person>();
        }
        if (sortBy == 1)
        {
            display_list = new LinkedList<Person>(display_list.OrderBy(x => x.getName()).ToList());
        }
        else if (sortBy == 2)
        {
            display_list = new LinkedList<Person>(display_list.OrderBy(x => x.getIdNumber()).ToList());
        }
        else if (sortBy == 3)
        {
            display_list = new LinkedList<Person>(display_list.OrderBy(x => x.getContactNumber()).ToList());
        }

        return display_list;
    }
    
}